src/vellum/workflows/runner/runner.py

```diff
@@ -232,6 +235,8 @@ def __init__(
                 # all of that data is redundant and is derivable. It also clearly communicates that
                 # there was no initial state provided by the user to invoke the workflow.
                 self._should_emit_initial_state = False
+                if execution_id is not None and previous_execution_id is None:
```

Comment:
DevinAI I think we want this if check to live at the end of this massive if/else check block and to only happen once right above where it says "# This queue is responsible for sending events from WorkflowRunner to the outside world"